pkg/deviceplugin/server.go

```diff
@@ -149,6 +159,7 @@ func (srv *server) Stop() error {
 	if srv.grpcServer == nil {
 		return errors.New("Can't stop non-existing gRPC server. Calling Stop() before Serve()?")
 	}
+	srv.state = terminating
```

Comment:
The unit test detects a race condition here. Could you please add accessor methods for writing and reading `server.state`? The methods need to be guarded with a mutex.

Reply:
Good tests! Will do.